### ISSUE TITLE
[DataGrid] - Possible EventEmitter memory leak detected

### ISSUE THREAD

**Issue by 913-Filp-Teodor** [mui-org/material-ui-x#530]
After using the datagrid I realised the following warning in the console:
![image](https://user-images.githubusercontent.com/30023610/97870874-45118d00-1d1c-11eb-8984-beb0055a6243.png)
I have 10 columns in my datagrid, and I would like to know how I can address this warning. I also noticed that the warning appears in your demos.

**Comment by 913-Filp-Teodor** [comment]
As far as I can see the warning appears only after having more than  7-8 columns in the datagrid (if I'm not mistaken)

**Comment by oliviertassinari** [comment]
@913-Filp-Teodor Could you try with this version?

```json
"@material-ui/data-grid": "https://pkg.csb.dev/mui-org/material-ui-x/commit/6e800398/@material-ui/data-grid"
```

We should have fixed this already in master;

**Comment by 913-Filp-Teodor** [comment]
Yes, it does work. But for the version 4.0.0-alpha.8 (current) it doesn't. Is this fix part of a future release then?

**Comment by oliviertassinari** [comment]
@913-Filp-Teodor Ok great, it will be fixed in the next release 4.0.0-alpha.9.

**Comment by oliviertassinari** [comment]
@DanailH I can't remember in which pull request this bug was fixed. Feel free to link the one that solved it you know.

**Comment by markedwards** [comment]
I'm encountering this issue with @mui/x-data-grid 5.1.0, when paging forward 3 times.  Has this regressed?

**Comment by m4theushw** [comment]
@markedwards Could you open another issue with a reproduction?

**Comment by TylerPop** [comment]
@markedwards I had this same issue and what fixed it for me was storing the array for my rows data in a useState and updating it inside of a useEffect. This stopped my rows from being set/updated on every render, which may have caused the memory leak and regardless is just bad for performance. This technique could probably be applied to both the rows and/or columns, if you have them updating frequently. Hope this helps!
